src/main/java/org/motechproject/prevac/service/impl/PrimeVaccinationScheduleServiceImpl.java

```diff
@@ -141,14 +147,13 @@ private Visit updateVisitWithDto(Visit primeDetails, Visit screeningDetails,
         }
     }
 
-    private void checkNumberOfPatients(PrimeVaccinationScheduleDto dto, Clinic clinic) {
-
+    private void checkNumberOfPatients(PrimeVaccinationScheduleDto dto, Clinic clinic) { //NO CHECKSTYLE CyclomaticComplexity
         List<Visit> visits = visitBookingDetailsDataService.findByPlannedDateClinicIdAndVisitType(dto.getDate(),
                 clinic.getId(), VisitType.PRIME_VACCINATION_DAY);
 
-        visitLimitationHelper.checkCapacityForVisitBookingDetails(dto.getDate(), clinic, dto.getVisitId());
-        if (visits != null) {
-            int numberOfRooms = clinic.getNumberOfRooms();
+        visitLimitationHelper.checkCapacityForVisit(dto.getDate(), clinic, dto.getVisitId());
+        if (visits != null && !visits.isEmpty()) {
+            int numberOfRooms = clinic.getNumberOfRooms() == null ? 0 : clinic.getNumberOfRooms();
```

Comment:
you cannot set it to 0, because if it's optional it should not be checked at all, remove this and change it to Integer and the last if to: "if (numberOfRooms != null && patients >= numberOfRooms)"